Sources/CSProgress/CSProgress.swift

```diff
@@ -600,7 +600,7 @@ public final class CSProgress: CustomDebugStringConvertible {
                     parent.sendFractionCompletedNotifications(fractionCompleted: parent.backing.fractionCompleted, isCompleted: parent.backing.isCompleted, completionHandler: completionHandler)
                 }
             }
-        } else if abs(fractionCompleted - lastNotifiedFractionCompleted) >= self.granularity {
+        } else if (isCompleted && lastNotifiedFractionCompleted != 1.0) || abs(fractionCompleted - lastNotifiedFractionCompleted) >= self.granularity {
```

Comment:
Do you have a minimal test case that can reproduce this issue? The change above relies on float inequality, which is generally not a good idea, but this may be equivalent:

```suggestion
        } else if abs(fractionCompleted - lastNotifiedFractionCompleted) >= self.granularity || isCompleted {
```

I am curious whether this solves the issue you were seeing.